ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/backwardsync/BackwardSyncContext.java

```diff
@@ -331,20 +334,9 @@ protected Void saveBlock(final Block block) {
   @VisibleForTesting
   protected void possiblyMoveHead(final Block lastSavedBlock) {
     final MutableBlockchain blockchain = getProtocolContext().getBlockchain();
-    if (maybeHead.isEmpty()) {
-      LOG.debug("Nothing to do with the head");
-      return;
-    }
-
-    final Hash head = maybeHead.get();
-    if (blockchain.getChainHead().getHash().equals(head)) {
-      LOG.debug("Head is already properly set");
-      return;
-    }
 
-    if (blockchain.contains(head)) {
-      LOG.debug("Changing head to {}", head);
-      blockchain.rewindToBlock(head);
+    if (blockchain.getChainHead().getHash().equals(lastSavedBlock.getHash())) {
```

Comment:
Instead of maybeHead, I am checking the passed in hash (which should originate from the FCU). I'm not sure if this check is useful at all though.

Reply:
You also removed the blockchain.contains(hash) check. Any reason for that? Could possiblyMoveHead be called before we save the block?

Reply:
Previously it was only acting on maybeHead and that is now removed. There wasn't an equivalent check for lastSavedBlock.

possiblyMoveHead is currently only called immediately after appending the block to the chain https://github.com/hyperledger/besu/blob/688d2dd3b4fe55fa9580c058666e16aabf102a6c/ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/backwardsync/BackwardSyncContext.java#L314-L317

I'm wondering whether to remove possiblyMoveHead entirely now as I'm not sure we would ever rewind to the last saved block (should always match this if condition I think)...
Would be good to get a second opinion from @fab-10

Reply:
Looking at the code `possiblyMoveHead` could be removed, but I am still if thinking if this method was introduced to manage cases when there was a reorg, and still do not have a full answer, anyway the amount of tests that you have done make me confident